src/torrent.cpp

```diff
@@ -1825,9 +1825,10 @@ namespace libtorrent
 		file_storage const& fs = m_torrent_file->files();
 		for (int i = 0; i < fs.num_files(); ++i)
 		{
+			if (!fs.pad_file_at(i) || fs.file_size(i) == 0) continue;
```

Comment:
need_picker() is not called and will crash when calling `picker->we_have` for padding files